packages/flutter/lib/src/painting/rounded_rectangle_border.dart

```diff
@@ -69,6 +70,13 @@ class RoundedRectangleBorder extends ShapeBorder {
         circleness: 1.0 - t,
       );
     }
+    if (a is StadiumBorder) {
+      return new _RoundedRectangleToStadiumBorder(
+        side: BorderSide.lerp(a.side, side, t),
+        borderRadius: borderRadius,
+        circleness: 1.0 - t,
+      );
+    }
```

Comment:
this shouldn't be needed, since StadiumBorder already knows how to do the reverse.

What was the error message that you saw without this?

Reply:
You are correct, the `_RoundedRectangleToStadiumBorder` wasn't needed. The actual problem was a mistake in the (outline_button.dart) `_OutlineBorder.lerpFrom` and `_OutlineBorder.lerpTo` classes.